File: src/main/java/com/drunkensoftball/api/bat/repo/AtBatRepository.java
package com.drunkensoftball.api.bat.repo;

import com.drunkensoftball.api.bat.domain.AtBat;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;

public interface AtBatRepository extends JpaRepository<AtBat, Long> {

    AtBat findByUuid(String uuid);
    AtBat findByUniqueIdAndGameIdAndRosterEntryId(String uniqueId, Long gameId, Long rosterId);

    List<AtBat> findByRosterEntryId(Long rosterId, Pageable pageable);
    List<AtBat> findByGameId(Long gameId, Pageable pageable);

}

File: src/main/kotlin/com/drunkensoftball/api/bat/repo/AtBatRepository.kt
package com.drunkensoftball.api.bat.repo


import com.drunkensoftball.api.bat.domain.AtBat
import org.springframework.data.domain.Pageable
import org.springframework.data.jpa.repository.JpaRepository

interface AtBatRepository : JpaRepository<AtBat, Long> {

    fun findByUuid(uuid: String): AtBat?
    fun findByUniqueIdAndGameIdAndRosterEntryId(uniqueId: String, gameId: Long?, rosterId: Long?): AtBat?

    fun findByRosterEntryId(rosterId: Long?, pageable: Pageable): List<AtBat>
    fun findByGameId(gameId: Long?, pageable: Pageable): List<AtBat>

}
